baseline/client/tests/n-back.test.js

```diff
@@ -1,20 +1,56 @@
 import { NBack } from "../n-back/n-back.js";
 import { pressKey, cartesianProduct } from "./utils.js"
 
+const pressBubblingKey = key => {
+    const initDict = {key: key, bubbles: true};
+    const display = document.querySelector(".jspsych-display-element");
+    display.dispatchEvent(new KeyboardEvent("keydown", initDict));
+    display.dispatchEvent(new KeyboardEvent("keyup", initDict));
+};
+
+const completeCurrentTrial = correctly => {
+    const trial = jsPsych.currentTrial();
+    const progress = jsPsych.progress();
+    if (trial.type === "html-keyboard-response") {
+        if (typeof trial.trial_duration === "number") {
+            jest.advanceTimersByTime(trial.trial_duration);
+        } else {
+            pressKey(trial.choices[0]);
+        }
+    } else if (trial.type === "n-back") {
+        trial.sequence.forEach((x, i) => {
+            if (correctly && (trial.n === 0 ? x === "1" : x === trial.sequence[i - trial.n])) {
+                pressBubblingKey(" ");
+            }
+            jest.advanceTimersByTime(trial.show_duration + trial.hide_duration);
+        });
+    }
+    // assert that progress occurred
+    if (jsPsych.progress().current_trial_global <= progress.current_trial_global) {
+        throw Error("progress didn't increase");
+    }
+};
+
 describe("n-back", () => {
     it("results should have at least one result marked isRelevant", () => {
```

Comment:
This test fails for me:

```
 FAIL  tests/n-back.test.js
  ● n-back › results should have at least one result marked isRelevant

    TypeError: jsPsych.data.get(...).filter(...).values(...).at is not a function

      116 |                     timeline: [this.constructor.simpleInstruction(train_instruction_retry_html)],
      117 |                     conditional_function: () => {
    > 118 |                         const trialData = jsPsych.data.get().filter({trial_type: "n-back"}).values().at(-1);
          |                                                                                                      ^
      119 |                         return !passingPracticeData(trialData);
      120 |                     }
      121 |                 }

      at Object.conditional_function (n-back/n-back.js:118:102)
      at TimelineNode.advance (node_modules/@adp-psych/jspsych/jspsych.js:542:58)
      at TimelineNode.advance (node_modules/@adp-psych/jspsych/jspsych.js:575:89)
      at TimelineNode.advance (node_modules/@adp-psych/jspsych/jspsych.js:575:89)
      at nextTrial (node_modules/@adp-psych/jspsych/jspsych.js:954:29)
      at Object.<anonymous>.window.jsPsych.core.finishTrial (node_modules/@adp-psych/jspsych/jspsych.js:339:9)
      at end_trial (node_modules/@adp-psych/jspsych/plugins/jspsych-html-keyboard-response.js:101:15)
      at node_modules/@adp-psych/jspsych/plugins/jspsych-html-keyboard-response.js:142:9
      at Object.callback (node_modules/@jest/fake-timers/build/legacyFakeTimers.js:625:32)
      at FakeTimers._runTimerHandle (node_modules/@jest/fake-timers/build/legacyFakeTimers.js:659:15)
      at FakeTimers.advanceTimersByTime (node_modules/@jest/fake-timers/build/legacyFakeTimers.js:294:14)
      at completeCurrentTrial (tests/n-back.test.js:16:18)
      at Object.<anonymous> (tests/n-back.test.js:48:13)
```

Reply:
Fixed by replacing `at(-1)` with `slice(-1)` and destructuring. (`Array.at` is too new!)